Fix:

CMake Error at cmake/ConnectorName.cmake:30 (ENDMACRO):
  Flow control statements are not properly nested.

https://github.com/HaxeFoundation/neko/pull/225

Included in:
https://github.com/HaxeFoundation/neko/commit/a88ed6bb14df859f8866091237b60da5cb330055
--- cmake/patch_mariadb.cmake.orig	2019-10-24 12:51:25.000000000 -0500
+++ cmake/patch_mariadb.cmake	2023-01-12 23:26:04.000000000 -0600
@@ -10,5 +10,17 @@
 	content ${content}
 )
 
+file(WRITE ${cmakelists} ${content})
+
+set(cmakelists ${mariadb_source}/cmake/ConnectorName.cmake)
+
+file(READ ${cmakelists} content)
+
+# Fix broken syntax on newer CMake
+string(REPLACE
+	"  END()"
+	"  ENDIF()"
+	content ${content}
+)
 
 file(WRITE ${cmakelists} ${content})
\ No newline at end of file
